### DISCUSSION TITLE
Struggling to backup stock firmware? Try without using any hubs!

### DISCUSSION THREAD

**Discussion by BenBergman** [emporia-vue-local/esphome#211] [General]
For those struggling with backing up their firmware, make sure your USB to serial adapter is connected directly to your computer and not through a USB hub. I had typed up the below message after struggling for hours with my board connected to the USB port on my monitor and then just before hitting submit I decided to try a port on the front of my desktop. Hopefully my misfortune can help someone else in the future.

> I've been struggling to backup the stock firmware for a while now. I'm running esptool.py on Linux with this command:
> 
> ```
> esptool.py --port /dev/ttyUSB0 -b 921600 read_flash 0 0x800000 flash_contents.bin
> ```
> 
> The first few times I ran this command it errored out with something about expecting 0x1000 bytes but only receiving 0xfe6 or something (sorry, didn't save the output). Other times it seems to hang without giving any hit of progress (the 0x1000 error typically showed something like 4% progress).
> 
> I then found [this thread](https://github.com/emporia-vue-local/esphome/discussions/31) that suggested shorting out the R82 resistor next to the RX pin. I have added such a jumper, but it is continuing to hang after establishing the baud rate.
> 
> ```
> esptool.py v4.7-dev
> Serial port /dev/ttyUSB0
> Connecting....
> Detecting chip type... Unsupported detection protocol, switching and trying again...
> Connecting...
> Detecting chip type... ESP32
> Chip is ESP32-D0WD (revision v1.0)
> Features: WiFi, BT, Dual Core, 240MHz, VRef calibration in efuse, Coding Scheme None
> Crystal is 40MHz
> MAC: ec:62:60:ca:a0:ac
> Uploading stub...
> Running stub...
> Stub running...
> Changing baud rate to 921600
> Changed.
> ```
> 
> Most recently I'm trying the git version of esptool, but originally I was trying with 4.6.2. I think it was the change to the dev version that changed my error from the 0x1000 bytes to the hang.
> 
> I have tried both 5V power on the 5V pin as separately tried 3.3V on the 3.3V pin with the same results. I've replaced the wires between my serial adapter (an FT232 board) in case the original ones were bad. I have a new CP2104 serial adapter in the mail that has the DSR/CTS/RTS pins which I'm hoping will help.
> 
> Is there anything else I could be trying?

**Comment by coolstuff99** [comment]
An alternative (it happened to me...) was to power the device with external 5V charger.
Problem was my usb was unable to provide sustainable 5V during backup process so ended up failing every time.
